File: Assignment 26 Feb Twitter live/AJAYKE/backend/app/models/book.py
class Book:
    def __init__(self, id, title, author, publication_year, genre, rating):
        self.id = id
        self.title = title
        self.author = author
        self.publication_year = publication_year
        self.genre = genre
        self.rating = rating

    def to_dict(self):
        return {
            "id": self.id,
            "title": self.title,
            "author": self.author,
            "publication_year": self.publication_year,
            "genre": self.genre,
            "rating": self.rating
        }

File: Assignment 26 Feb Twitter live/Aman123at/src/types/book.ts
export interface BookMetadata {
  pages: number;
  stockLeft: number;
  price: number;
  discount: number;
  edition: number;
}

export interface Book {
  id: string;
  title: string;
  author: string;
  publicationYear: number;
  genre: string;
  rating: number;
  description: string;
  metadata: BookMetadata;
}

export interface Statistics {
  averageRatingByGenre: Record<string, number>;
  oldestBook: Book;
  newestBook: Book;
} 